src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -48,4 +49,14 @@ public List<Book> findAll() {
             throw new RuntimeException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book b where b.id = :id", Book.class)
+                    .setParameter("id", id).stream().findFirst();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get book by id: " + id, e);
```

Comment:
Use EntityNotFound exception here

Reply:
`EntityNotFoundException` should be thrown here